### PR TITLE
Use poetry for packaging

### PR DESCRIPTION
**Pull request by sjagoe**
Part one of addressing #196

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+quiet = "haas.plugins.result_handler:QuietTestResultHandler"
+verbose = "haas.plugins.result_handler:VerboseTestResultHandler"
+timing = "haas.plugins.result_handler:TimingResultHandler"
+
+[tool.poetry.dependencies]
+python = "^3.7"
+statistics = "^1.0.3.5"
+stevedore = [
+    { version = "^3.5.2", python = "<3.8" },
+    { version = "^4.1.1", python = ">=3.8" }
```
Comment (itziakos) [inline pyproject.toml:52]: Do we really need to be so strict in the runtime dependencies? Stevedore is stable enough and I think it is save to have ">=4.1.1"

**Review by itziakos — APPROVED** [review]
I am not sold on poetry but I can see the appeal. On thing that bugs me is that one needs poetry packages to build the package from the sdist. Is there a way that the generated sdist uses just setuptools without any dynamic features?

File: pyproject.toml (hunk 1 above)
Comment (itziakos) [reply pyproject.toml:52]: FYI, locally I build haas using stevedore 5.x

Comment (sjagoe) [reply pyproject.toml:52]: It's typical in application development to pin dependencies to semantic-version compatible. I'm open to either. Or to adding v5 as the version used for later python versions.

**Comment by sjagoe** [comment]
> I am not sold on poetry

What options do we have that satisfy these requirements?
1. Use pyproject.toml exclusively (with plugin entry points as well)
2. Support generating the version number from git and writing it to the source

I didn't look at many, but chose poetry for its metadata format being more pleasant to use than `hatchling`

**Comment by sjagoe** [comment]
Okay, I'm not up-to-date on new setuptools. The only think I see missing I think is the dynamic version number, but otherwise we could use that instead of poetry. Although I kind of like poetry for the reason of it managing the virtualenv for me

**Comment by sjagoe** [comment]
I don't see a way for using pyproject.toml for development dependencies with setuptools. It looks like going that way would lead back to a mix of different dependency management solutions (requirements files for some subset) rather than defining all project metadata in pyproject.toml.

To me poetry is the clearly more complete solution.

**Comment by itziakos** [comment]
@sjagoe please go on with the PR as we discussed.

**Comment by sjagoe** [comment]
Closed in favour of #199

**Pull request closed without merging**